src_c/geometry.c

```diff
@@ -8,6 +8,106 @@
 
 #define PYGAMEAPI_GEOMETRY_NUMSLOTS 21
 
+static PyObject *
+pg_raycast(PyObject *_null, PyObject *const *args, Py_ssize_t nargs)
+{
+    PyObject **farr;
+    Py_ssize_t farr_length;
```

Comment:
I know i did it myself, but we should rename all `farr` thingies into more readable things, like i would suggest you rename `farr` -> `colliders` and `farr_length` -> `colliders_length`